### ISSUE TITLE
Bug:  The `CreateChatCompletionStream` do not  return ` Rate limit ` and  ` Quota exceed `  error correctly.

### ISSUE THREAD

**Issue by guoqiangqi** [sashabaranov/go-openai#250]
The `CreateChatCompletionStream` do not return ` Rate limit` and `Quota exceed` error correctly, which is important for user to retry their requests.
```
stream, err := client.CreateChatCompletionStream(
		context.Background(),
		openai.ChatCompletionRequest{
			Model:    model,
			Messages: messages,
			Stream:   true,
		},
	)
```

I always get the `err` with `nil` value when actually meet a ` Rate limit` and `Quota exceed` issue, ` client.CreateChatCompletion` works well.

**Comment by guoqiangqi** [comment]
Hi @nasa1024 , i notice that you have fixed some issues about `chatCompletionStream`, can you do me a favor? Really appreciate it!

**Comment by sashabaranov** [comment]
I would also recommend starting with a test where mock HTTP server returns 429 error code

**Comment by guoqiangqi** [comment]
Hi @sashabaranov , I noticed that this issue has been fixed in #259 . Do you have any plans to add this patch to a specific release version, so that I can use it?

**Comment by guoqiangqi** [comment]
> Hi @sashabaranov , I noticed that this issue has been fixed in #259 . Do you have any plans to add this patch to a specific release version, so that I can use it?

Sorry , i see this in [v1.9.0](https://github.com/sashabaranov/go-openai/releases/tag/v1.9.0), thanks.
